src/libraries/System.Private.Xml/src/System/Xml/Serialization/CodeGenerator.cs

```diff
@@ -1538,11 +1539,12 @@ public bool TryGetValue(string key, out LocalBuilder value)
             }
         }
 
-        public LocalBuilder this[string key]
+        [DisallowNull]
```

Comment:
The setter doesn't have an explicit null check here right? Is the `DisallowNull` attribute still correct?

Reply:
yes, if you assign null here then `_locals` second generic arg will be nullable which will cause chain reaction everywhere as nothing expects that to be null... This is an internal class so not a good tradeoff 😄